### ISSUE TITLE
signing RPMs with sha256 headers/payloads built with nfpm is not supported 

### ISSUE THREAD

**Issue by ll78912** [sassoftware/go-rpmutils#28]
Currently, this module only works for signing RPMs that have md5/sha1 digests. The output below was generated with `rpm -Kv` to inspect RPMs  built with fpm vs nfpm respectively.
```
    Header SHA1 digest: OK
    MD5 digest: OK
```
However, RPMs built with nfpm are with sha256 digests:
```
  Header SHA256 digest: OK
  Payload SHA256 digest: OK
```
Are there any possibilities to extend this module to support RPMs that are built with nfpm?

**Comment by elliotpeele** [comment]
Can you provide a sample rpm that could be used for testing? Preferably something that only contains one file and can be used for test data.

**Comment by David78912** [comment]
[nfpm.zip](https://github.com/sassoftware/go-rpmutils/files/12446501/nfpm.zip)
@elliotpeele: thanks for the reply. I have uploaded the RPM built with nfpm for your testing purpose. You can also run the following commands to regenerate the RPM:
```
 nfpm package --config test.yaml --packager rpm 
```
Please let me know if you need anything additional.

**Comment by David78912** [comment]
[@elliotpeele](https://github.com/elliotpeele): thanks for the reply. I have uploaded the RPM built with nfpm for your testing purpose: [nfpm.zip](https://github.com/sassoftware/go-rpmutils/files/12446552/nfpm.zip). You can also rerun the following commands to regenerate the RPM:
 nfpm package --config test.yaml --packager rpm 

Please let me know if you need anything additional.

**Comment by ll78912** [comment]
@elliotpeele: thanks for your reply earlier. Just checking to see if there are plans to support such feature?

**Comment by ll78912** [comment]
@elliotpeele: Thanks for working on this issue. I posted a comment within in the PR that the new logic for v4 signatures is not compatible with `rpm -k` command while the old logic does support this for md5 digests. I am just wondering if there are plans to support such feature? Thanks!

**Comment by ll78912** [comment]
@elliotpeele : any updates on this issue? thank you

**Comment by mtharp** [comment]
Fixed in v0.3.0, let me know if it doesn't work for you.
